internal/logging/logger.go

```diff
@@ -2,205 +2,104 @@
-//
-// SetServerLogger should not be called concurrently. It should be called
-// before any goroutines that may use the logger are started.
-func SetServerLogger() {
-	L = newServerLogger(level, os.Stdout, os.Stderr)
-	S = L.Sugar()
 }
 
-func newServerLogger(level zapcore.LevelEnabler, stdout, stderr zapcore.WriteSyncer) *zap.Logger {
-	if term.IsTerminal(int(os.Stdin.Fd())) {
-		writer := zapcore.Lock(filtered(stderr))
-		encoder := zapcore.NewConsoleEncoder(zapcore.EncoderConfig{
-			MessageKey: "message",
-
-			LevelKey:    "level",
-			EncodeLevel: zapcore.CapitalColorLevelEncoder,
-
-			TimeKey:    "time",
-			EncodeTime: zapcore.ISO8601TimeEncoder,
-
-			CallerKey:    "caller",
-			EncodeCaller: zapcore.ShortCallerEncoder,
-		})
-
-		return zap.New(zapcore.NewCore(encoder, writer, level), zap.AddCaller())
+func NewLogger(writer io.Writer) *logger {
+	return &logger{
+		zerolog.New(writer).With().Timestamp().Caller().Logger(),
 	}
-
-	return zap.New(
-		zapcore.NewCore(
-			zapcore.NewJSONEncoder(zap.NewProductionEncoderConfig()),
-			zapcore.Lock(filtered(stdout)),
-			level,
-		),
-		zap.AddCaller(),
-	)
 }
 
-func NewFileErrorLog(outputFilePath string) *zap.Logger {
-	w := zapcore.AddSync(&lumberjack.Logger{
-		Filename:   outputFilePath,
-		MaxSize:    10, // megabytes
-		MaxBackups: 0,
-		MaxAge:     28, // days
-	})
-
-	core := zapcore.NewCore(
-		zapcore.NewJSONEncoder(zap.NewProductionEncoderConfig()),
-		w,
-		zap.ErrorLevel,
-	)
-
-	return zap.New(core)
+func UseServerLogger() {
+	L = NewLogger(os.Stderr)
 }
 
-func StandardErrorLog() *log.Logger {
-	errorLog, err := zap.NewStdLogAt(L, zapcore.ErrorLevel)
-	if err != nil {
-		return nil
-	}
-
-	return errorLog
+func (l *logger) Tracef(format string, v ...interface{}) {
+	l.Trace().Msgf(format, v...)
 }
 
-// TODO: Remove the filtered writer after Go stops writing header
-// values to errors, as it's cpu expensive to search every log line.
-// https://github.com/golang/go/pull/48979
-func filtered(logger zapcore.WriteSyncer) zapcore.WriteSyncer {
-	return &filteredWriterSyncer{
-		dest: logger,
-	}
+func (l *logger) Debugf(format string, v ...interface{}) {
+	l.Debug().Msgf(format, v...)
 }
 
-type filteredWriterSyncer struct {
-	dest zapcore.WriteSyncer
+func (l *logger) Infof(format string, v ...interface{}) {
+	l.Info().Msgf(format, v...)
 }
 
-var strInvalidHeaderFieldValue = []byte("invalid header field value")
-
-func (w *filteredWriterSyncer) Write(b []byte) (int, error) {
-	if idx := bytes.Index(b, strInvalidHeaderFieldValue); idx >= 0 {
-		idx += len(strInvalidHeaderFieldValue)
-
-		forKeyIdx := bytes.Index(b, []byte("for key"))
-		if forKeyIdx > idx {
-			return w.dest.Write(append(b[:idx+1], b[forKeyIdx:]...))
-		}
+func (l *logger) Warnf(format string, v ...interface{}) {
+	l.Warn().Msgf(format, v...)
+}
 
-		if b[0] != '{' {
-			// not json; free to truncate.
-			return w.dest.Write(b[:idx])
-		}
+func (l *logger) Errorf(format string, v ...interface{}) {
```

Comment:
Should we document these compatibility helpers as deprecated? I suspect we probably want to encourage everyone to use the `zerolog.Logger` methods directly so that we can add context fields. And  we probably want to avoid having yet another way of logging (since zerolog seems to have at least a couple ways already)

Reply:
Possibly. I like this simplified interface since the `zerolog` API requires a `Msg` to finalize the log, it's easy to forget and end up with nothing being logged. There's definitely scenarios where we'd want to use the `zerolog` API directly, particularly for structured logging or contexts but that might be overkill for most cases